### PR TITLE
Added lang attribute for accessibility and search-engine parsers

### LINKED ISSUE #2844: Shiny HTML element has no lang attribute
### System details
- Shiny 1.4.0.9002
- R 3.6.3

### Repro Steps
- Create a simple Shiny App following tutorials or use the default template created in the RStudio IDE (New Project / Shiny Web Application)
- Run the app and load into browser (I'm using Chrome on macOS)
- Examine the HTML or run an accessibility validation tool such as Axe (https://www.deque.com/axe/axe-for-web/)

### Problem
The HTML element has no `lang` attribute. This is a WCAG 2.1 A violation: https://www.w3.org/TR/WCAG21/#language-of-page

### Expected
Pages should be accessible by default. Thus, there should be a reasonable default, and the ability to specify a different language. For English:

`<html lang="en">`

This is critical for screen readers, otherwise you can have (for example) a Spanish user's screen reader trying to pronounce English words using Spanish language rules.

There was a community question on this:
https://community.rstudio.com/t/how-to-specify-the-language-to-a-shiny-app/26472

### PR DESCRIPTION
**Pull request by schloerke**
Fixes #2844 
Fixes #2920

#2920 main comment:
> * Added the `lang` argument to `shinyApp()` that specifies document-level language within the app for the accessibility of screen readers and search-engine parsers.
> * Users can pass [ISO 639-1 language code](https://www.w3schools.com/tags/ref_language_codes.asp) for this argument.
> * Defaulting to `lang = ""` when no input is given by users. I was pondering the default  lang value set to either "en" or auto-locale-detect method; however, that can lead to unexpected issues. For example, many people use English R and locale while creating shiny apps in their own languages. We cannot assume only based upon users' locale which can lead to other accessibility side effects for screen readers. Thus, defaulting to empty string is a safer way.
>
> \- `@jooyoungseo`

Followup changes to #2920: 

* Removed if statements
* Assign lang unconditionally
* Removed `lang` from `basicPage()` args
* Use `@inheritParams` for `@param lang`
* Added helper method to create the `html` start tag. 
  * Avoids `<html lang="">` when lang is `NULL` or empty string.

### CODE REVIEW

File: R/shinyui.R [hunk 1]
```diff
@@ -38,7 +40,10 @@ renderPage <- function(ui, showcase=0, testMode=FALSE) {
     # Put the body into the default template
     ui <- htmlTemplate(
       system.file("template", "default.html", package = "shiny"),
-      body = ui
+      lang = lang,
+      body = ui,
+      # this template is a complete HTML document
+      document_ = TRUE
```
Comment (wch) [inline R/shinyui.R:46]: Just checking: the template previously was autodetected so that this behaved like it was TRUE, right?

File: inst/template/default.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html{{ if (isTRUE(nzchar(lang))) paste0("lang=\"", lang, "\"") }}>
```
Comment (wch) [inline inst/template/default.html:2]: Just checking: if `lang` is NULL, would any extra spaces be inserted here?

File: R/shinyui.R (hunk 1 above)
Comment (schloerke) [reply R/shinyui.R:46]: Correct. I'll be removing this line though

File: inst/template/default.html (hunk 2 above)
Comment (schloerke) [reply inst/template/default.html:2]: Nope

```
❯❯ isTRUE(nzchar(NULL))
[1] FALSE
```

File: tests/testthat/test-test.R [hunk 3]
```diff
@@ -161,7 +161,7 @@ test_that("app template works with runTests", {
     )
   ))
 
-  lapply(combos, function(combo) {
+  for (combo in combos) {local({
```
Comment (wch) [inline tests/testthat/test-test.R:164]: Just curious: why the change?

File: inst/template/default.html (hunk 2 above)
Comment (schloerke) [reply inst/template/default.html:2]: And everything between the brackets (and including the brackets) will be removed

**Review by wch — APPROVED** [review]
Looks good! I just had a few small questions.

File: tests/testthat/test-test.R (hunk 3 above)
Comment (schloerke) [reply tests/testthat/test-test.R:164]: `testthat` says that the failure occurs at line 164 and not 172 or the actual failure line

File: R/shinyui.R (hunk 1 above)
Comment (schloerke) [reply R/shinyui.R:46]: I'm going to keep this line in.

Prior behavior had `document_ = "auto"` by default.  It would hit these lines of code in `htmltools`: https://github.com/rstudio/htmltools/blob/f589c77bc941428d4bf2cec3ca5f15399ce18f47/R/template.R#L63-L65 where `htmltools` would test for some form of `<html>` to set `document_`.

To make sure the `lang` logic does not corrupt the template `document_` logic, I am explicitly setting `document_ = TRUE`.

File: inst/template/default.html (hunk 2 above)
Comment (wch) [reply inst/template/default.html:2]: Sounds good - I was mostly interested in making sure that the `{{ }}` didn't result in any extra spaces.
